client/options.go

```diff
@@ -66,6 +67,12 @@ func ForceDowngrade(force bool) ConnectOption {
 	return forceDowngradeOption(force)
 }
 
+// OverriddenContentType returns a connection option that instructs the
+// client to use a custom content type for sending requests to the server.
+func OverriddenContentType(contentType string) ConnectOption {
```

Comment:
How about naming it more aligned with the functional options pattern?

```suggestion
func WithContentType(contentType string) ConnectOption {
```

Reply:
Thanks, changed to `WithContentType`